source/scpp/src/scp/QuorumSetUtils.cpp

```diff
@@ -33,24 +34,50 @@ class QuorumSetSanityChecker
     bool mIsSane;
     size_t mCount{0};
 
-    bool checkSanity(SCPQuorumSet const& qSet, int depth);
+    bool checkSanity(SCPQuorumSet const& qSet, int depth, const char** reason);
 };
 
 QuorumSetSanityChecker::QuorumSetSanityChecker(SCPQuorumSet const& qSet,
-                                               bool extraChecks)
+                                               bool extraChecks,
+                                               const char** reason)
     : mExtraChecks{extraChecks}
 {
-    mIsSane = checkSanity(qSet, 0) && mCount >= 1 && mCount <= 1000;
+    const char* msg;
+    if (reason == nullptr)
+        reason = &msg;  // avoid null checks in checkSanity()
+
+    mIsSane = checkSanity(qSet, 0, reason);
+    if (mCount < 1)
+    {
+        const char* msg = "Number of validator nodes is zero";
```

Comment:
Am I missing something w.r.t. the `const char* msg = ...; *reason = msg` pattern ? Wouldn't `*reason = ...` be more straightforward ?
Also assert that `reason` and `mIsSane` are consistent at the end of this constructor.